smartsim/_core/utils/telemetry/telemetry.py

```diff
@@ -355,20 +140,27 @@ def init_launcher(self, launcher: str) -> Launcher:
 
         raise ValueError("Launcher type not supported: " + launcher)
 
-    def set_launcher(self, launcher_type: str) -> None:
-        """Set the launcher for the experiment"""
-        self._launcher = self.init_launcher(launcher_type)
+    def set_launcher(self, launcher: str) -> None:
+        """Set the launcher for the experiment
+
+        :param launcher: type of launcher used by the experiment
+        :type launcher: str"""
+        self._launcher = self.init_launcher(launcher)
         self.job_manager.set_launcher(self._launcher)
         self.job_manager.start()
 
     def process_manifest(self, manifest_path: str) -> None:
-        """Read the runtime manifest for the experiment and track new entities
+        """Read the manifest for the experiment. Process the
+        `RuntimeManifest` by updating the set of tracked jobs
+        and registered collectors
 
-        :param manifest_path: The full path to the manifest file
-        :type manifest_path: str
-        """
+        :param manifest_path: full path to the manifest file
+        :type manifest_path: str"""
         try:
-            manifest = load_manifest(manifest_path)
+            # it is possible to read the manifest prior to a completed
+            # write due to no access locking mechanism. log the issue
+            # and continue. it will retry on the next event loop iteration
+            manifest = RuntimeManifest.load_manifest(manifest_path)
             if not manifest:
                 return
```

Comment:
Good question. I originally had one, but felt it might log a ton of messages for no "good" reason until a manifest is created.... I am adding it back in so we can better track what's happening during debugging.
```
            manifest = RuntimeManifest.load_manifest(manifest_path)
            if not manifest:
                logger.debug("No manifest file exists")
                return
```